test/extended/util/docker.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import (
+	dockerClient "github.com/fsouza/go-dockerclient"
+	tutil "github.com/openshift/origin/test/util"
+	"errors"
+)
+
+func TagImage(tagee, tagor string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.TagImageOptions{
+		Repo:  tagee,
+		Tag:   "latest",
+		Force: true,
+	}
+	terr := client.TagImage(tagor, opts)
+	return terr
+}
+
+func PullImage(name string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.PullImageOptions{
+		Repository: name,
+		Tag:        "latest",
+	}
+	perr := client.PullImage(opts, dockerClient.AuthConfiguration{})
+	return perr
+}
+
+func GetImageIDForTags(comps []string) ([]string, error) {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return nil, dcerr
+	}
+	imageList, serr := client.ListImages(dockerClient.ListImagesOptions{})
+	if serr != nil {
+		return nil, serr
+	}
+
+	returnTags := make([]string, 0)
+	for _, comp :=  range comps {
+		var found bool
+		for _, image := range imageList {
+			for _, repTag := range image.RepoTags {
+				if repTag == comp {
+					found = true
+					returnTags = append(returnTags, image.ID)
```

Comment:
So just to be sure I understand, let me try to reiterate in my own words:
1) the method should only return an error if an error returned from getting the docker client or the docker list images cmd has a problem
2) otherwise, just return a sparse array, with "" for the id's which are missing, but no err object
3) then the caller to this method looks for id's of "" instead of the err != nil as the indication that something went wrong and abort